game/player.hh

```diff
@@ -9,9 +9,11 @@
 #include <vector>
 #include <utility>
 
+#include <boost/optional.hpp>
+
 class Song;
 
-/// player class
+/// TODO: rename to Vocalist or something like that
```

Comment:
Correct me if i'm wrong but a `Player` is not persé a vocalist it can also be guitar-player (guitar hero mode) or a dancer (ddr-mode) i believe.
If the latter is the case you can revert this comment

Reply:
No, it seems it's exactly Vocalist/Singer: it processes microphone audio stream. As far as I understand dancers and guitar players are instances of InstrumentGraph class :)

Reply:
Alright can you create another issue with details on how to refactor this to Vocalist/Singer. I prefer `Singer` If you have added the issue feel free to remove that comment again :)

Reply:
@Baklap4 and here is an issue: #543 :)